aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -56,13 +60,17 @@
     // local storage adapter, and a remote API
     private final Orchestrator orchestrator;
 
+    // Keeps track of whether of not the category is initialized yet
+    private final CountDownLatch categoryInitializationsPending;
```

Comment:
Minor nit but maybe since this is effectively the latch variable we're using for concurrency management in category initialization we could have explicit naming of `categoryInitializationLatch`?

Reply:
The naming here is awkward, agreed. The logic here was that it's a count that starts at 1 and then goes to 0, so it's like, how can I name this _count_ in a reasonable way? _Quantity of things that still need some operation..._, etc.